File: client/src/api/model/target.ts
export type Target = typeof Target[keyof typeof Target];

export const Target = {
  Self: 'Self',
  Opponent: 'Opponent',
  'Engaged_Ally': 'Engaged Ally',
  'Engaged_Enemy': 'Engaged Enemy',
  'Any_Ally': 'Any Ally',
  'Any_Enemy': 'Any Enemy',
} as const;

File: server/src/main/java/com/github/genraven/genesys/domain/enums/Target.java
package com.github.genraven.genesys.domain.enums;

import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Schema(enumAsRef = true)
@AllArgsConstructor
@Getter
public enum Target {
    /** Applies to the performer themselves. */
    SELF("Self"),
    /** Applies to a direct opponent or roll made against this character. */
    OPPONENT("Opponent"),
    /** An ally who is at engaged range. */
    ENGAGED_ALLY("Engaged Ally"),
    /** An enemy who is at engaged range. */
    ENGAGED_ENEMY("Engaged Enemy"),
    /** Any ally within range. */
    ANY_ALLY("Any Ally"),
    /** Any enemy within range. */
    ANY_ENEMY("Any Enemy");

    @JsonValue
    private final String label;
}
